### PR TITLE
Added a link under "Unit not found" to view all services

### LINKED ISSUE #16946: services: Unit not found pages have no action
Generally, you probably won't hit a unit not found page. But in bugs like #16945, you may. And there may be some edge cases where you could hit it, especially for user services which might be spawned just for one session and not for another.

Right now, the page looks like this:

image

We should use an [empty state pattern](https://www.patternfly.org/v4/components/empty-state/) with an action. The action would just be a link back to the main page. (It should be styled like a link, not a button, as it goes somewhere instead of doing something.)

---

**TL;DR**: Add a link under "Unit not found" that says "View all services" which goes back to the services list.

### PR DESCRIPTION
**Pull request by Zach-Clay**
Screenshot from 2022-02-21 09-53-02

Fixes #16946

### CODE REVIEW

File: pkg/systemd/services/services.jsx [hunk 1]
```diff
@@ -729,9 +729,17 @@ class ServicesPageBody extends React.Component {
             const unit_path = get_unit_path(unit_id);
 
             if (unit_path === undefined) {
-                return <EmptyStatePanel
+                return (<EmptyStatePanel
```
Comment (marusak) [inline pkg/systemd/services/services.jsx:732]: Are these `()` necessary here?

File: pkg/systemd/services/services.jsx [hunk 2]
```diff
@@ -729,9 +729,17 @@ class ServicesPageBody extends React.Component {
...
                             icon={ExclamationCircleIcon}
-                            title={_("Unit not found")} />;
+                            title={_("Unit not found")}
+                            secondary={
```
Comment (marusak) [inline pkg/systemd/services/services.jsx:735]: Using `secondary` seems like the most logical solution. However it feels like there is too much space now.
`secondary`:
Screenshot from 2022-02-11 07-29-24

`paragraph`:
paragraph

WDYT?

**Review by marusak — DISMISSED** [review]
Awesome job, thank you!
It would be great if you could add test for it. You can read about how to run tests locally [here](https://github.com/cockpit-project/cockpit/blob/main/test/README.md). You most likely want to add test into `test/verify/check-system-services`. Just going to nonexisting service, clicking the button and making sure it leads to services page would be enough. Thanks!

While on re-pushing, please also `--amend` commit message to use [imperative mood](https://cbea.ms/git-commit/#imperative), add `services:` prefix and add [Fixes #...](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) line into commit message.

**Comment by Zach-Clay** [comment]
@marusak Hi! I have removed the parentheses that were not needed also also made the link a `paragraph` instead of `secondary`. I think this removes some of the unwanted space. 

In terms of adding a test, I took a look at the file and I am not really sure how to go about doing this...

**Comment by marusak** [comment]
> While trying to run the test I am getting multiple errors on top of my VM crashing

If you want to, please post your issues here or join us on IRC (#cockpit on libera) and we can help you

> how important is it that I write a test in order for my PR to be merged?

It won't land without tests. But if you prefer, I am happy to write tests and push them here.

File: pkg/systemd/services/services.jsx [hunk 3]
```diff
@@ -731,7 +731,14 @@ class ServicesPageBody extends React.Component {
             if (unit_path === undefined) {
                 return <EmptyStatePanel
                             icon={ExclamationCircleIcon}
-                            title={_("Unit not found")} />;
+                            title={_("Unit not found")}
+                            paragraph={
+                                <Button variant="link"
+                                        component="a"
+                                        onClick={() => cockpit.jump("/system/services", cockpit.transport.host)}>
```
Comment (marusak) [inline pkg/systemd/services/services.jsx:738]: I just realized, that when you would see this for user unit, you should go to all user units and not to system ones

**Comment by Zach-Clay** [comment]
@marusak 
> It won't land without tests. But if you prefer, I am happy to write tests and push them here.

That would be awesome given my current workload!

**Comment by marusak** [comment]
Rebased, cleaned up commits, added tests and fixed redirection link to work correctly for system and user. Running tests now.

**Review by jelly — APPROVED** [review]

**Pull request merged**